### DISCUSSION TITLE
How to configure `JsonOptions` for controllers?

### DISCUSSION THREAD

**Discussion by hieucd04** [OrchardCMS/OrchardCore#12623] [Q&A] [answered]
I'm migrating from `Newtonsoft.Json` to `System.Text.Json` and my controllers now refuse to bind its models because of incorrect default `JsonOptions`;

Usually, this should work:
```csharp
services.AddControllers().AddJsonOptions(options =>
{
    options.JsonSerializerOptions.PropertyNameCaseInsensitive = true;
    options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
});
```

However, as far as I remember, OrchardCore doesn't allow calling `AddControllers()`.

I tried this approach but it doesn't work:

```csharp
public static void ConfigureServices(IServiceCollection services)
{
    services.AddOrchardCms().AddSetupFeatures("OrchardCore.AutoSetup");

    // I tried putting it everywhere: in application, in a module, etc -> not working!
    services.Configure<JsonOptions>(options =>
    {
        options.JsonSerializerOptions.PropertyNameCaseInsensitive = true;
        options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
    });
}
```

**Comment by ns8482e** [comment, marked as answer]
@hieucd04 OC uses Newtonsoft.Json

**Reply by hieucd04** [reply]
Good to know!
I removed `Newtonsoft.Json` from the project including the `CamelCasePropertyNamesContractResolver` -> that explains why my controllers are no longer working.
